src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.ts

```diff
@@ -404,6 +422,42 @@ export class ChallengeleaderboardComponent implements OnInit, AfterViewInit {
       data => {
         SELF.updateLeaderboardResults(data['results'], SELF);
         SELF.startLeaderboard(SELF.selectedPhaseSplit['id']);
+        SELF.sortLeaderboardTextOption = SELF.showLeaderboardByLatest ? 'Sort by best' : 'Sort by latest';
+      },
+      err => {
+        SELF.globalService.handleApiError(err);
+      },
+      () => {}
+    );
+  }
+
+  showLeaderboardByLatestOrBest() {
+    const API_PATH = this.endpointsService.particularChallengePhaseSplitUrl(this.selectedPhaseSplit['id']);
+    const SELF = this;
+    let BODY;
+
+    SELF.showLeaderboardByLatest = !SELF.showLeaderboardByLatest;
```

Comment:
It might happen that API gets call return with some error so it we should not update this variable here instead we should update this variable too after the success of API call.

Reply:
Changed!